### PR TITLE
Adding MenuBar #68

### PR DESCRIPTION
**Pull request by kikilondon**
#68

### CODE REVIEW

File: frontend/src/components/MenuBar/MenuBar.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: "row",
+    columnGap: 80,
+    justifyContent: "center",
+    alignItems: "center",
+    padding: 20,
+    backgroundColor: "#A9A9A9" // I cannot set this color as globalStyleVariable.js background
```
Comment (pablisch) [inline frontend/src/components/MenuBar/MenuBar.js:29]: To do this, choose an appropriates name, e.g. menubarBackground and use that
`backgroundColor: styleVariables.menubarBackground`
And add that colour to the global variables,
`export const styleVariables = {
  backgroundColour: "#f5f5f5",
  unselected: "lightgrey",
  singleSelected: "lightblue",
  doubleSelected: "darkturquoise",
  menubarBackground: "#A9A9A9"
};`

File: frontend/package.json [hunk 2]
```diff
@@ -9,11 +9,9 @@
     "web": "expo start --web",
     "prettier-format": "prettier --config .prettierrc 'src/**/*.js' --write",
     "lint:fix": "npx eslint . && npm run prettier-format"
-
   },
   "dependencies": {
     "@expo/vector-icons": "^14.0.0",
-    "@react-navigation/native": "^6.1.10",
```
Comment (pablisch) [inline frontend/package.json:16]: This was in main previous to your branch and presumably is used somewhere. Easiest thing to do is run `npm install @react-navigation/native` to reinstall it.

File: frontend/src/components/Button/ButtonStyles.js [hunk 3]
```diff
@@ -3,14 +3,14 @@ import { StyleSheet } from "react-native";
 export const styles = StyleSheet.create({
   container: {
     elevation: 8,
-    backgroundColor: "#2E8B57",
```
Comment (pablisch) [inline frontend/src/components/Button/ButtonStyles.js:6]: I think it is probably best to revert all changes to the button component for now. It was not intended as a part of this ticket to add any buttons as they will need to be displayed by context. Buttons will always need to be different sizes, colours, etc. so we will have to look at how to pass these in as props to this component.

File: frontend/src/components/MenuBar/MenuBar.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import { useNavigation } from "@react-navigation/native";
+import { View, Text, TouchableHighlight } from "react-native";
+import { styleVariables } from "../../styles/globalStyleVariables";
+import Button from "../Button/Button";
+
+export default function MenuBar() {
+  const navigation = useNavigation();
+  return (
+    <View style={styles.container}>
+      <Button title="Clear all" />
```
Comment (pablisch) [inline frontend/src/components/MenuBar/MenuBar.js:16]: As mentioned, this was not a part of this ticket but great to have them here, just don't worry about the styling.

File: frontend/src/components/MenuBar/MenuBar.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { StyleSheet } from "react-native";
+//import { TouchableOpacity, Text } from "react-native";
```
Comment (pablisch) [inline frontend/src/components/MenuBar/MenuBar.js:3]: There are a lot of comments and unused code, e.g. TouchableHighlight and Text are imported but not used. In general, unless there is some reason, I would get rid of unused code as when left, nobody knows why it is there and it tends to stay forever.

**Review by pablisch — CHANGES_REQUESTED** [review]
It was a confusingly small ticket. 
Worth implementing the colour variable now I think although thinking about, probably best to stick with Ju's colour unless you have some reason for changing it. I'm not sure about the green on the plan personally but we should assume that the main screen will have some colour, or maybe not? Might be worth a discussion?

Please replace the missing dependency and delete any code you don't want. If you are sure it is needed then you can leave a comment about it.

If any comments are confusing, please ask.

File: package.json [hunk 6]
```diff
@@ -16,5 +16,8 @@
   "license": "ISC",
   "devDependencies": {
     "husky": "^9.0.11"
+  },
```
Comment (pablisch) [inline package.json:19]: This was installed in the root folder of the project, so should be removed from here where there should be no dependencies, and be installed in the frontend instead.

File: frontend/src/components/MenuBar/MenuBar.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { StyleSheet } from "react-native";
+import { useNavigation } from "@react-navigation/native";
+import { View, Text, TouchableHighlight } from "react-native";
```
Comment (pablisch) [inline frontend/src/components/MenuBar/MenuBar.js:4]: Do we need Text and TouchableHighlight here? I don't imagine we will use Text here and I have no clue what TouchableHighlight is.

Navigation is also imported and declared but never used. If you think we will use it later that is fine.

**Review by pablisch — CHANGES_REQUESTED** [review]
Almost there!

**Comment by kikilondon** [comment]
@pablisch I removed react-navigation/native dependency  from the package.json root folder and moved to frontend folder. 

I don't know anything about husky dependency, maybe there is a specific reason why they installed it there.If not, please let me know, and I will move it to the package.json front-end folder.

**Comment by pablisch** [comment]
> Member

Don't worry about husky. I don't know what it is but it is something Sidra is doing about pre-commit hooks which is all a mystery to me. I imagine it relates to FE and BE so is in the root. I should have been clearer about meaning dependencies as opposed to dev-dependencies. 

I see that @react-navigation/native is back in the FE but I also still see it in the root. Since there should be no dependencies there all of the selected code here should go.

Screenshot 2024-03-13 at 12 28 50

If that is not showing on your local branch, check you have pushed everything (I keep forgetting this!) and if it's not that, let me know and we'll sort it.

Everything else looks ready to go.

**Review by pablisch — APPROVED** [review]
Great!

**Comment by kikilondon** [comment]
Great ! Finally 🤣🤣

**Pull request merged**